core/src/main/java/org/n52/iceland/ogc/gml/time/TimePosition.java

```diff
@@ -62,8 +54,7 @@ public TimePosition(DateTime time) {
      *            Indeterminate value of time position
      */
     public TimePosition(TimeIndeterminateValue indeterminateValue) {
-        super();
-        this.indeterminateValue = indeterminateValue;
+        this.indeterminateValue = Optional.of(indeterminateValue);
```

Comment:
Then create two different classes.

I can create a `TimePosition` with an indeterminate value and can change that to any non-`null` value. I also can create a `TimePosition` with a `DateTime` and can change it's `DateTime` to any non-`null` value. But I can't create a `TimePosition` and change it's type from indeterminate value to `DateTime`. That's two classes for me.

Alternatives are that you make `TimePosition` immutable, or that you make `TimePosition` fully mutable... But in the last case you'll still can have the inconsistent state you have now, e.g.:
```java
new TimePosition(time); // throws NPE
TimePostion t1 = new TimePosition(DateTime.now(), TimeFormat.ISO8601);
t2.setIndeterminateValue(TimeIndeterminateValue.now); // all fields filled
TimePostion t2 = new TimePosition(TimeIndeterminateValue.now);
t2.setTime(DateTime.now()); // missing TimeFormat
TimePosition t3 = new TimePostition(TimeIndeterminateValue.now);
t3.setTimeFormat(TimeFormat.YMD); // missing Time
```